odoo/addons/base/models/res_device.py

```diff
@@ -0,0 +1,184 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from contextlib import nullcontext
+from datetime import datetime
+import logging
+
+from odoo import api, fields, models, tools
+from odoo.http import GeoIP, request, root
+from odoo.tools import SQL
+from odoo.tools.translate import _
+from .res_users import check_identity
+
+_logger = logging.getLogger(__name__)
+
+
+class ResDeviceLog(models.Model):
+    _name = 'res.device.log'
+    _description = 'Device Log'
+    _rec_name = 'display_name'
+
+    session_identifier = fields.Char("Session Identifier", required=True, index='btree')
+    platform = fields.Char("Platform")
+    browser = fields.Char("Browser")
+    ip_address = fields.Char("IP Address")
+    country = fields.Char("Country")
+    city = fields.Char("City")
+    device_type = fields.Selection([('computer', 'Computer'), ('mobile', 'Mobile')], "Device Type")
+    user_id = fields.Many2one("res.users", index='btree')
+    first_activity = fields.Datetime("First Activity")
+    last_activity = fields.Datetime("Last Activity", index='btree')
+    revoked = fields.Boolean("Revoked", index='btree',
```

Comment:
In fact, the index is useless in this case because the number of `False` values is much greater than the number of `True` values and this field is only used in a `where` condition to filter with `False` values, postgresql will not use the index. Thanks for this comment !